cognite/client/data_classes/simulators/simulators.py

```diff
@@ -0,0 +1,517 @@
+            if "unitQuantities" in resource
+            else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.model_types, SimulatorModelType):
+            output["modelTypes" if camel_case else "model_types"] = self.model_types.dump(camel_case=camel_case)
+        if isinstance(self.step_fields, SimulatorStep):
+            output["stepFields" if camel_case else "step_fields"] = self.step_fields.dump(camel_case=camel_case)
+        if isinstance(self.unit_quantities, SimulatorQuantity):
+            output["unitQuantities" if camel_case else "unit_quantities"] = self.unit_quantities.dump(
+                camel_case=camel_case
+            )
+
+        return output
+
+
+class SimulatorWrite(SimulatorCore):
+    def __init__(
+        self,
+        external_id: str,
+        name: str,
+        file_extension_types: str | SequenceNotStr[str],
+        model_types: SimulatorModelType | Sequence[SimulatorModelType] | None = None,
+        step_fields: SimulatorStep | Sequence[SimulatorStep] | None = None,
+        unit_quantities: SimulatorQuantity | Sequence[SimulatorQuantity] | None = None,
+    ) -> None:
+        super().__init__(
+            external_id=external_id,
+            name=name,
+            file_extension_types=file_extension_types,
+            model_types=model_types,
+            step_fields=step_fields,
+            unit_quantities=unit_quantities,
+        )
+
+    def as_write(self) -> SimulatorWrite:
+        """Returns a writeable version of this resource"""
+        return self
+
+
+class Simulator(SimulatorCore):
+    def __init__(
+        self,
+        external_id: str,
+        name: str,
+        file_extension_types: str | SequenceNotStr[str],
+        created_time: int | None = None,
+        last_updated_time: int | None = None,
+        id: int | None = None,
+        model_types: SimulatorModelType | Sequence[SimulatorModelType] | None = None,
+        step_fields: SimulatorStep | Sequence[SimulatorStep] | None = None,
+        unit_quantities: SimulatorQuantity | Sequence[SimulatorQuantity] | None = None,
+    ) -> None:
+        self.external_id = external_id
+        self.name = name
+        self.file_extension_types = file_extension_types
+        self.model_types = model_types
+        self.step_fields = step_fields
+        self.unit_quantities = unit_quantities
+        # id/created_time/last_updated_time are required when using the class to read,
+        # but don't make sense passing in when creating a new object. So in order to make the typing
+        # correct here (i.e. int and not Optional[int]), we force the type to be int rather than
+        # Optional[int].
+        self.id: int | None = id
+        self.created_time: int | None = created_time
+        self.last_updated_time: int | None = last_updated_time
```

Comment:
When you add both a read and a write version, _then this is not needed_. Just make these required in the init signature (I'll add a separate comment about it)
```suggestion
        self.id = id
        self.created_time = created_time
        self.last_updated_time = last_updated_time
```